### PR TITLE
Tax is calculated twice for virtual products if Apple/Google Pay is used on product pages

### LINKED ISSUE #7577: Tax is calculated twice for virtual products if Apple/Google Pay is used on product pages
### Describe the bug
With certain tax settings, tax is calculated twice for virtual products if Apple/Google Pay is used on product pages.

### To Reproduce

Settings:

Screenshot 2023-10-29 at 13 45 52

Screenshot 2023-10-29 at 13 46 07

Screenshot 2023-10-29 at 13 46 55

Product settings:

Screenshot 2023-10-29 at 13 48 53

1. Open a browser that works with Apple or Google Pay set up.
2. Click the Apple or Google Pay button on the product page.
3. The tax is already included into the product price but it's added again in the Apple/Google Pay window on the product page:

Screenshot 2023-10-29 at 13 50 02

If a product is not virtual, then tax is added for a couple of seconds and then removed:

Screenshot 2023-10-29 at 13 52 15
Screenshot 2023-10-29 at 13 52 24

Also, this doesn't happen on cart or checkout pages.

### Actual behavior

Tax is added on top of the product price. If product is virtual, that price with 2x tax remains displayed in Apple/Google Pay window, while if a product is not virtual - it's only displayed for a few seconds and then correct price is displayed.

### Expected behavior

Tax is already included in product price so shouldn't be included twice.

### Desktop (please complete the following information):

* OS: 13.4.1
* Browser [e.g. chrome, safari] Safari
* Version [e.g. 22] 118.0.5993.88 

### Additional context

7231902-zen

### PR DESCRIPTION
**Pull request by gpressutto5**
Fixes #7577

#### Changes proposed in this Pull Request

- This PR renames the method `prices_exclude_tax` to `cart_prices_include_tax` and invert its logic to make the code easier to read.
- It also adds two tests to ensure tax calculation is correct and consistent for product pages and cart details
- The key fix was changing `$product->get_price()` to `wc_get_price_excluding_tax()`
- WooPay class was also updated for consistency

#### Testing instructions

1. Setup Apple Pay in your environment. pdjVC1-11t-p2
1. Go to WooCommerce >> Settings >> Tax and set the settings from the original issue:
	1. Yes, I will enter prices inclusive of tax
	2. Shop base address (for consistency, https://github.com/Automattic/woocommerce-payments/issues/7577#issuecomment-1847967971)
	3. Display prices in shop **Including taxes**
	4. Display prices during cart and checkout **Excluding taxes**
1. Add a Standard tax rate
2. Try to use Apple Pay and ensure the price is correct

- Additionally, run other scenarios from the spreadsheet from https://github.com/Automattic/woocommerce-payments/issues/7577#issuecomment-1847967971 and ensure everything works as expected.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7818` or branch name `fix-apple-pay-including-tax` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix-apple-pay-including-tax) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 2b1ad373dd66f089eed5d35dedeac0d960040be1
- Build time: 2023-12-14 14:33:54 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-payment-request-button-handler.php [hunk 1]
```diff
@@ -1549,7 +1549,7 @@ protected function build_shipping_methods( $shipping_methods ) {
 	 *
 	 * @param boolean $itemized_display_items Indicates whether to show subtotals or itemized views.
 	 */
-	protected function build_display_items( $itemized_display_items = false ) {
+	public function build_display_items( $itemized_display_items = false ) {
```
Comment (brettshumaker) [inline includes/class-wc-payments-payment-request-button-handler.php:1552]: Why the change from `protected` to `public` here?

**Review by brettshumaker — APPROVED** [review]
Everything works as described and tests well.

One question about the change from a protected method to a public one, but other than that, this looks good.

File: includes/class-wc-payments-payment-request-button-handler.php (hunk 1 above)
Comment (gpressutto5) [reply includes/class-wc-payments-payment-request-button-handler.php:1552]: Because I wanted to unit test it and have `$itemized_display_items = true`.

**Pull request merged**